elasticsearch/json-adaptive-selection-6plus.yml

```diff
@@ -0,0 +1,65 @@
+type: json
+data:
+  url: /_nodes/_local/stats/indices,transport,http,thread_pool,jvm,process,fs,script,discovery,ingest,adaptive_selection?format=smile
+  server: ${ST_MONITOR_ES_NODE_HOSTPORT}
+  basicHttpAuthUsername: ${ST_MONITOR_ES_NODE_BASICAUTH_USERNAME}
+  basicHttpAuthPassword: ${ST_MONITOR_ES_NODE_BASICAUTH_PASSWORD}
+  smileFormat: true
+require:
+  - className: com.sematext.spm.client.es.EsVersionCheck
+    values: 6-*
+
+observation:
+  - name: adaptive_replica_selection
+    metricNamespace: es
+    path: $.nodes.${nodeId}.adaptive_selection.${remoteNodeId}
+
+    metric:
+      - name: adaptiveReplicaSelection.searches.outgoing
+        source: outgoing_searches
+        type: long_gauge
+        label: outgoing searches
+        description: outgoing searches
+
+      - name: adaptiveReplicaSelection.queue.size.avg
+        source: avg_queue_size
+        type: double_gauge
+        label: average queue size
+        description: average queue size
+
+      - name: adaptiveReplicaSelection.service.time.avg
```

Comment:
Service time is the time it takes for a task in the thread pool to actually be executed. This is different from response time, which would include how long the task stayed in the queue. I will document this in the metric description, as you suggested in the other comments.

As for cumulative value, we don't have it :( Here's what we have, in the stats of each node (i.e. its own perspective), looking at each node's IDs:

```
      "adaptive_selection" : {
        "PLKGmqocSYSbs_vSGjUSVg" : {
          "outgoing_searches" : 0,
          "avg_queue_size" : 0,
          "avg_service_time_ns" : 251904,
          "avg_response_time_ns" : 1647092,
          "rank" : "1.6"
        },
        "8VdsJY-ATmGIjHHEj-aYhA" : {
          "outgoing_searches" : 0,
          "avg_queue_size" : 0,
          "avg_service_time_ns" : 564962,
          "avg_response_time_ns" : 1529914,
          "rank" : "1.5"
        }
```

Reply:
Uh, that sucks :(.

Reply:
It looks like it doesn't suck that badly, since what ES cares about (when computing the rank) is the exponentially weighted moving average. So we don't really have an average there.

Resolving this conversation for now, but feel free to reopen.

Reply:
It does suck badly :), but nothing we can do about it.